tests/test_admin/test_view.py

```diff
@@ -153,6 +154,58 @@ def test_missing_object(self):
         assert result.status_code == 404
 
 
+@mark.django_db
+@override_settings(ROOT_URLCONF="test_utils.admin_urls")
+class TestCatalogQueryPreviewView(TestCase):
+    """
+    Test the Catalog Query Preview view
+    """
+    def setUp(self):
+        """
+        Set up testing variables
+        """
+        self.user = UserFactory.create(is_staff=True, is_active=True)
+        self.user.set_password("QWERTY")
+        self.user.save()
+        self.client = Client()
+        self.catalog_query = EnterpriseCatalogQuery.objects.create(
+            title='Test Catalog Query',
+            content_filter={'partner': 'MuiX'}
+        )
+        super().setUp()
+
+    @mock.patch('enterprise.api_client.discovery.CourseCatalogApiServiceClient')
+    def test_preview_query(self, mock_catalog_api_client):
+        """
+        Test that we render the query preview correctly.
+        """
+        mock_catalog_api_client.return_value = mock.Mock(
+            get_catalog_results=mock.Mock(
+                return_value=fake_catalog_api.FAKE_SEARCH_ALL_RESULTS
+            ),
+        )
+        assert self.client.login(username=self.user.username, password="QWERTY")
+        url = reverse(
+            'admin:' + enterprise_admin.utils.UrlNames.PREVIEW_QUERY_RESULT,
+            args=(self.catalog_query.pk,)
+        )
+        result = self.client.get(url)
+        content_count = json.loads(result.content)['count']
+        self.assertEqual(content_count, 3)
+
+    def test_missing_object(self):
+        """
+        Test that a missing template object causes a 404.
+        """
+        assert self.client.login(username=self.user.username, password="QWERTY")
```

Comment:
Can we add a test for a non-admin and non-staff users, to make sure they don't have access to this API